src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -538,6 +548,16 @@ public async Task<ReturnResult> RenameAsync(IStorageItemWithPath source, string
 			var progress = new Progress<FileSystemProgress>();
 			progress.ProgressChanged += (s, e) => returnStatus = returnStatus < ReturnResult.Failed ? e.Status!.Value.ToStatus() : returnStatus;
 
+			if (string.IsNullOrWhiteSpace(newName)
+				|| ContainsRestrictedCharacters(newName)
+				|| ContainsRestrictedFileName(newName))
+			{
+				await DialogDisplayHelper.ShowDialogAsync(
+					"ErrorDialogThisActionCannotBeDone".GetLocalizedResource(),
+					"ErrorDialogNameNotAllowed".GetLocalizedResource());
+				return ReturnResult.Failed;
+			}
```

Comment:
Is it worth creating a common method for this code?

Reply:
The return types change, so I think we should leave it as is